### ISSUE TITLE
Coherent style

### ISSUE THREAD

**Issue by dweindl** [AMICI-dev/AMICI#887]
At some point, we should give the AMICI API a makeover. Things like the mix of camelcase and snakecase in arguments of the same function are annoying/confusing. Also parameters referring to the same entity would ideally have same name when occurring in different function signatures.

Also: In C++ we are using camelcase, in python snakecase. This leads to a funny mix in the Python API. "Deal with it" or do we want to make that a bit more coherent at some point (i.e. snakecase for C++)?

**Comment by FFroehlich** [comment]
Agreed. I would vote for having snakecase in C++ (requires a bit of work though).

**Comment by paulstapor** [comment]
So, the vote is for snakecase in C++?
I'm asking because I'm writing a new spline implementation in C++ right now, and would be happy to directly stick to the correct naming... :)

**Comment by dweindl** [comment]
> So, the vote is for snakecase in C++?

Mine is for https://google.github.io/styleguide/cppguide.html

**Comment by paulstapor** [comment]
Okay, looks like a rather extensive style guide... Totally fine for me to fix this one as a reference...

**Comment by dweindl** [comment]
I would consider addressing the CamelCase / snake_case issue in v1.0.0 - shall we go for a rough transition here or keep the CamelCase versions as aliases for another while (maybe only for the Python interface - not sure if the C++ interface is currently used anywhere)? I'd currently vote for a rough transition.
